### PR TITLE
DOC: Add PredictionResults to documentation

### LINKED ISSUE #3753: DOC: Prediction Results, Contrast Results, where in the docs
do we have a docstring for the instance returned by get_prediction, e.g. GLM ?

there is no see also in the get_prediction docstring http://www.statsmodels.org/dev/generated/statsmodels.genmod.generalized_linear_model.GLMResults.get_prediction.html

I guess similar for contrast results returned by t_test, ...

### PR DESCRIPTION
**Pull request by bashtage**
Add PredictionResults to documentation
Import Results in parent class to all documentation from public mod

close #3753

### CODE REVIEW

File: statsmodels/regression/linear_model.py [hunk 1]
```diff
@@ -52,11 +52,12 @@
 from statsmodels.tools.sm_exceptions import InvalidTestWarning
 
 # need import in module instead of lazily to copy `__doc__`
+from statsmodels.regression._prediction import PredictionResults
 from . import _prediction as pred
```
Comment (jbrockmendel) [inline statsmodels/regression/linear_model.py]: Any reason not to include this in the following line?

**Comment by bashtage** [comment]
The as and use of the same name make it difficult.

File: statsmodels/regression/linear_model.py (hunk 1 above)
Comment (josef-pkt) [reply statsmodels/regression/linear_model.py]: the second import is for the module

**Comment by josef-pkt** [comment]
LGTM, merging

general issue: AFAIR, this is the first import into the main module api of a method results class. 
In this case the underlying module is private. We will have to check what we do for other method results classes, e.g. OLSInfluence get_influence, ContrastResults and margins are in public module and are or can be documented there.  We will most likely have more cases over time

**Pull request merged**